frontend/src/filter_table.tsx

```diff
@@ -0,0 +1,218 @@
+          {rowData.number}
+        </a>
+    },
+    {field: 'sex', title: 'Kön', hidden: false},
+    {field: 'birth_date', title: 'Födelsedatum', hidden: false},
+    {field: 'death_date', title: 'Dödsdatum', hidden: false},
+    {field: 'death_note', title: 'Dödsanteckning', hidden: false},
+    {field: 'mother', title: 'Moder', hidden: false,
+      render: (rowData:any) =>
+      <a className={styles.functionLink}
+        onClick={() => popup(<IndividualView id={rowData.mother['number']} />, `/individual/${rowData.mother['number']}`)}
+        >
+        {rowData.mother['number']}
+      </a>
+    },
+    {field: 'father', title: 'Fader', hidden: false,
+      render: (rowData:any) =>
+      <a className={styles.functionLink}
+        onClick={() => popup(<IndividualView id={rowData.father['number']} />, `/individual/${rowData.father['number']}`)}
+        >
+        {rowData.father['number']}
+      </a>
+    },
+    {field: 'color', title: 'Färg', hidden: false,
+      render: (rowData:any) => rowData.color['name']
+    },
+    {field: 'color_note', title: 'Färganteckning', hidden: false},
+  ]
+
+  const [columns, setColumns] = React.useState(defaultColumns)
+  const [currentFilters, setFilters] = React.useState(filters)
+
+  const updateColumns = (values: any[]) => {
+    const newColumns = [...columns];
+    const visibleColumns = values.map((v: any) => v.value);
+    newColumns.forEach((c: any) => {
+      if (visibleColumns.includes(c.field)) {
+        c.hidden = false;
+      } else {
+        c.hidden = true;
+      }
+    })
+    setColumns(newColumns)
+  }
+
+  const filteredIndividuals = () => {
+    if (!individuals) {
+      return []
+    }
+    return individuals.filter((i: Individual) => {
+      for (let f = 0; f < currentFilters.length; f++) {
+        if (currentFilters[f].active && !i[currentFilters[f].field]) {
+          return false;
+        }
+      }
+      return true
+    })
+  }
+
+  return <>
+    <div>
+
+      <FormControlLabel
+        className={styles.columnLabel}
+        label="Kolumner:"
+        labelPlacement="start"
+        control={
+          <Select className={styles.columnSelect}
+            isMulti
+            options={columns.map((v: any) => {return {value: v.field, label: v.title}})}
+            value={columns.filter((v: any) => !v.hidden).map((v: any) => {return {value: v.field, label: v.title}})}
+            onChange={updateColumns}
+            />
+        }
+      />
+    </div>
+    <div className={styles.table}>
+      { individuals
+        ? <>
+            {currentFilters.map(filter =>
+              <FormControlLabel key={filter.field}
+                control={<Checkbox
+                          name={filter.field}
+                          checked={filter.active}
+                          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                            filter.active = e.target.checked
+                            setFilters([...currentFilters])
```

Comment:
`currentFilters` is already the filter! This is the same as `setFilters(currentFilters)`, which is the same as not setting! 

If you want to get around the mutation you can do something like 
```typescript
const next = [...currentFilters]
next[i] = {...next[i], active: e.target.checked } // i index in the map
setFilters(next)
```
:man_shrugging:

Reply:
I think it works due to the spread operator! `a = []; a != [...a]`.